### ISSUE TITLE
Dockerize

### ISSUE THREAD

**Issue by 0xdabbad00** [duo-labs/cloudmapper#287]
Enough people run into issues getting it up and running I need to look into running it via Docker in a smart way that lets you take advantage of all the commands, not just drawing a network diagram like the previous PR's have focused on.

**Comment by FernandoMiguel** [comment]
How can I help?
Haven't touched this in months, so not sure what the inputs and outputs are anymore

**Comment by 0xdabbad00** [comment]
@FernandoMiguel One question I have is whether it makes sense to try to pass commands and parameters to the docker container somehow for running each different command, or if I should just assume people will ssh into the docker container and have an easy way for them to get data in and out of the docker environment.  I'm leaning toward the "just ssh in" model, as I don't even know if there is another way of doing things.

Another concern I have is around the data collection and can Docker containers expand as the data within it grows?  For example, a recent engagement had 8GB of metadata collected from their accounts.

**Comment by FernandoMiguel** [comment]
There's no ssh, just exec.
Not that the pedantic me is gone, 
Most people expect to be able to run commands from their host

**Comment by FernandoMiguel** [comment]
Docker storage is limited by how much you set up on the agent.
For example, Mac defaults to 60GB if I'm not mistaken, but you can increase it to the total of free disk. 


Also, you can just mount the storage output to the host disk, and that removes that concern

**Comment by 0xdabbad00** [comment]
Looks like somehow `numpy` version `1.16.1` get's installed even though the Pipfile specifically requires `1.15.1`.   Trying to run cloudmappy results in the error:
```
ImportError: Something is wrong with the numpy installation. While importing we detected an older version of numpy in ['.../virtualenvs/cloudmapper-M_tGLtdm/lib/python3.7/site-packages/numpy']. One method of fixing this is to repeatedly uninstall numpy until none is found, then reinstall this version.
```

So looks like I need to focus more on ensuring exactly what I've requested is installed, probably using sha hashes.  I was not expecting dependencies to be this much of a problem for this project.

**Comment by 0xdabbad00** [comment]
Doing `pip uninstall numpy` seems to fix this issue for me as somehow the correct version of numpy is also installed, and that uninstall command only removes the 1.16.1 version.  This is a mess.

**Comment by Salihan04** [comment]
Hi. Just saw this issue after trying to run the demo from a Docker container. Currently I am able to get "CloudMapper serving on 127.0.0.1:8000" but when I access it from the browser I get a "localhost didn't send any data". Any advice? Thanks

**Comment by 0xdabbad00** [comment]
In a quick skim, this looks incorrect @Salihan04 https://gist.github.com/Salihan04/b500ba06627f36d1fa0993c87bcb047a#file-docker-compose-yml-L9

Until there is an official docker image for this though, docker related issues are going to be out of scope for what I'm willing to support questions on as too many things can go wrong with people's own docker configs.

**Comment by livehybrid** [comment]
@Salihan04 Are you attempting to access it on port 9000 in your browser?

**Comment by livehybrid** [comment]
@Salihan04 Are you accessing it via port 9000 in your browser?

**Comment by Salihan04** [comment]
> @Salihan04 Are you attempting to access it on port 9000 in your browser?

yes

**Comment by Salihan04** [comment]
> @Salihan04 Are you accessing it via port 9000 in your browser?

yes

**Comment by 0xdabbad00** [comment]
PRs for where Docker support was added include #54 and #27, and apparent fix for something in #106.  Removed docker support in #153.  PR that tried to add it back in #221

**Comment by 0xdabbad00** [comment]
Docckerfile added back in #494.  I expect how this is used to change in the future, as the primary use case is for #483
